packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -585,7 +585,7 @@ function renderWeightedTargets(weightedTargets: WeightedTarget[]): CfnRoute.Weig
   for (const t of weightedTargets) {
     renderedTargets.push({
       virtualNode: t.virtualNode.virtualNodeName,
-      weight: t.weight || 1,
+      weight: t.weight == undefined ? 1 : t.weight,
```

Comment:
Shouldn't this be `0`?

Reply:
I agree with @ytsssun  This should be `weight: t.weight || 0`. Is there a reason we are defaulting to 1?

Reply:
0 would be a better default because [The total weight for all targets combined must be less than or equal to 100. ](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-route-weightedtarget.html). But I wanted to keep this PR as small as possible without changing the established defaults.

It might be that changing this default could break something for some users. @TheRealAmazonKendra what do you think?